### DISCUSSION TITLE
Segment Anything Model (SAM)

### DISCUSSION THREAD

**Discussion by tolahs** [ultralytics/ultralytics#7044] [Docs]
Hi everybody!
I want to ask sth. for [this link  topic given below](https://docs.ultralytics.com/models/sam/#sam-comparison-vs-yolov8)
*Auto-Annotation: A Quick Path to Segmentation Datasets*

What is the correct way choose if i have pre-trained model from created my custom dataset ?

Use my custom detection model(my_custom_detection_model)  or use yolov pretrained model(yolov8x.pt) ?

`auto_annotate(data="path/to/images", det_model="yolov8x.pt", sam_model='sam_b.pt')`
`auto_annotate(data="path/to/images", det_model="my_custom_detection_model.pt", sam_model='sam_b.pt')`

**Comment by pderrenger** [comment]
@tolahs hello! If you have a pre-trained custom detection model that is well-tuned for your specific dataset, it would generally be more beneficial to use that for auto-annotation, as it is likely to produce more accurate detection results tailored to your data. This can lead to better quality segmentation datasets created by the SAM model. So, in your case, you might want to use:

```python
auto_annotate(data="path/to/images", det_model="my_custom_detection_model.pt", sam_model='sam_b.pt')
```

However, if your custom model is not performing as well as you'd like, or if it hasn't been trained on a sufficiently diverse dataset, you might find that the pre-trained `yolov8x.pt` model offers better detection performance out-of-the-box, which could also improve the auto-annotation results.

It's a good practice to experiment with both options and compare the quality of the auto-annotated datasets to determine which detection model works best for your needs. 😊👍
